cmd/run.go

```diff
@@ -125,6 +125,16 @@ a commandline interface for interacting with it.`,
 				return err
 			}
 
+			// Parse the thresholds, only if the --no-threshold flag is not set.
+			if !runtimeOptions.NoThresholds.Bool {
+				for _, thresholds := range conf.Options.Thresholds {
+					err := thresholds.Parse()
+					if err != nil {
+						return errext.WithExitCodeIfNone(err, exitcodes.InvalidConfig)
+					}
+				}
+			}
+
```

Comment:
Unfortunately I think we need to copy this to `cmd/cloud.go` and `cmd/archive.go` :thinking: Looking at the code the alternative is to add it in `getConsolidatedConfig`, which arguably will make this more testable :thinking:

Reply:
I've added the check to `cmd/cloud.go` and `cmd/archive.go`. I've added tests to verify the behavior is respected in the context of the `run` command. Looking into doing the same for `cloud` and `archive`. 

I chose not to put the code in `getConsolidatedConfig` as the actual check is really a side effect, one that does not have any impact on the config struct `getConsolidateConfig` produces (as far as I can tell). Moreover, I do think that validation before running will justify refactoring this somehow in the close future anyway.